### PR TITLE
Add table statistics functions for show table status

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr adds statistics functionality to dolt tables that allows `SHOW TABLE STATUS` to render data length.

### CODE REVIEW

File: go/go.sum [hunk 1]
```diff
@@ -148,16 +148,26 @@ github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZm
 github.com/dgryski/go-sip13 v0.0.0-20181026042036-e10d5fee7954/go.mod h1:vAd38F8PWV+bWy6jNmig1y/TA+kYO4g3RSRF0IAv0no=
 github.com/dolthub/fslock v0.0.2 h1:8vUh47iKovgrtXNrXVIzsIoWLlspoXg+3nslhUzgKSw=
 github.com/dolthub/fslock v0.0.2/go.mod h1:0i7bsNkK+XHwFL3dIsSWeXSV7sykVzzVr6+jq8oeEo0=
+<<<<<<< HEAD
```
Comment (andrew-wm-arthur) [inline go/go.sum]: you can usually resolve conflicts here by deleting the file and running `go mod init`

File: go/libraries/doltcore/sqle/tables.go [hunk 2]
```diff
@@ -275,6 +275,42 @@ func (t *DoltTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
...
+	for _, col := range schema {
+		switch n := col.Type.(type) {
+		case sql.NumberType:
+			numBytesPerRow += 8
+		case sql.StringType:
+			numBytesPerRow += uint64(n.MaxByteLength())
+		case sql.BitType:
+			numBytesPerRow += 1
+		case sql.DatetimeType:
+			numBytesPerRow += 8
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/tables.go:290]: timestamps are 16 bytes: https://github.com/golang/go/blob/e31e84010e0989e1df4bd2c55d529497119bb345/src/time/time.go#L127

File: go/libraries/doltcore/sqle/indexed_dolt_table.go [hunk 3]
```diff
@@ -152,6 +152,42 @@ func (t *WritableIndexedDoltTable) NumRows(ctx *sql.Context) (uint64, error) {
 	return m.Len(), nil
 }
 
+func (t *WritableIndexedDoltTable) DataLength(ctx *sql.Context) (uint64, error) {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/indexed_dolt_table.go:155]: `WritableIndexedDoltTable` embeds a `DoltTable`, so you shouldn't need to reimplement this

File: integration-tests/bats/sql-show.bats [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    dolt sql -q "CREATE TABLE test(pk int NOT NULL AUTO_INCREMENT, c1 int, PRIMARY KEY (pk))"
+
+    run dolt sql -q "show table status where \`Data_length\`=0"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "test" ]] || false
+
+    dolt sql -q "INSERT INTO test (c1) VALUES (0)"
+   run dolt sql -q "show table status where \`Data_length\`>0"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "test" ]] || false
```
Comment (andrew-wm-arthur) [inline integration-tests/bats/sql-show.bats:63]: we should assert an actual data length value for at least one of these tests

**Review by andrew-wm-arthur — APPROVED** [review]
LGMT, a few comments

**Comment by VinaiRachakonda** [comment]
Need to make some fixes to gms to get these tests working..

**Pull request merged**